### ISSUE TITLE
Add url request event in agent

### ISSUE THREAD

**Issue by jbfuzier** [refractionPOINT/limacharlie#50]
Hi,

It would be great if the agent was able to log http queries (url, method, headers...). Especially when a proxy server is used, there is no DNS request done by the endpoint and the IP connection seen if from the endpoint to the proxy server. That means that there is a blindspot here.

**Comment by maximelb** [comment]
This is definitely high in our priority list. The DNS is actually available on Windows, but it's worth noting it's gathered from the Windows DNS Resolver Cache, which means an executable doing its own UDP sockets resolutions wouldn't be caught, so it's not quite as precise, but it's definitely there.

The new solution for DNS will parse DNS responses off the wire on Mac and Windows. HTTP headers is likely going to be the same way, but considering the lack of HTTPS introspection it's not as high on the list. I'm toying with the idea of adding support for HTTP(S) headers as a Chrome extension that would get dropped by LC.

**Comment by jbfuzier** [comment]
What about hooking http/https related functions in Windows? [Such as those.](https://github.com/cuckoosandbox/monitor/blob/master/sigs/network.rst)
That would still mean some blind spots if http requests are done using plain TCP sockets but you will still see those in network connections.

**Comment by maximelb** [comment]
True, maybe at some point. The issue I had right now with that approach is that resources are limited (manpower) and doing hooking would only work on Windows (while basic parsing of HTTP would be cross platform), and hooking adds more complexity, especially if some AVs start detecting LC as malicious software. :-/

But yeah, tradeoffs. DNS is definitely high on the stack of TODOs.

**Comment by jbfuzier** [comment]
I understand. On the bright side, your sniffing idea allows to cover http with and without proxy and https limited to the domain in case of https over proxy. The DNS will give the domain in case of https.

Have you found a way to link sniffed traffic to a process ?

**Comment by maximelb** [comment]
Although I have not implemented it yet, it shouldn't be an issue. I should be able to extend the mechanism we use to track the connections to also collect the content bytes, so that would give us Traffic<-->PID, which will also be linked with Atoms internally to be included in the nice graph. :)
